node_modules/oae-principals/lib/internal/dao.js

```diff
@@ -285,6 +289,107 @@ var deletePrincipal = module.exports.deletePrincipal = function(principalId, cal
 };
 
 /**
+ * Marks a principal as being deleted. Currently only groups are supported
+ *
+ * @param  {String}     principalId     The id of the principal to mark as deleted
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occurred, if any
+ */
+var deletePrincipal2 = module.exports.deletePrincipal2 = function(ctx, user, callback) {
```

Comment:
This method needs a better name... Maybe `fullyDeletePrincipal` or something like that?